nixos/modules/services/misc/autosuspend.nix

```diff
@@ -0,0 +1,291 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.services.autosuspend;
+
+  general =
+    filterAttrs
+    (_: v: v != null)
+    (removeAttrs cfg ["enable" "package" "checks" "wakeups"]);
+  checks =
+    mapAttrs'
+    (n: v: {
+      name = "check.${n}";
+      value = filterAttrs (_: v: v != null) v;
+    })
+    cfg.checks;
+  wakeups =
+    mapAttrs'
+    (n: v: {
+      name = "wakeup.${n}";
+      value = filterAttrs (_: v: v != null) v;
+    })
+    cfg.wakeups;
+
+  autosuspend-conf =
+    pkgs.writeText
+    "autosuspend.conf"
+    (generators.toINI {} ({inherit general;} // checks // wakeups));
+
+  autosuspend = cfg.package;
+
+  version = "4.3.0";
+
+  checkType = with types;
+    submodule {
+      freeformType = with types; attrsOf anything;
+
+      options.enabled = mkOption {
+        default = false;
+        type = bool;
+        description = mdDoc ''
+          Whether to execute this activity check.
+        '';
+      };
+
+      options.class = mkOption {
+        default = null;
+        type = nullOr (enum [
+          "ActiveCalendarEvent"
+          "ActiveConnection"
+          "ExternalCommand"
+          "JsonPath"
+          "Kodi"
+          "KodiIdleTime"
+          "LastLogActivity"
+          "Load"
+          "LogindSessionsIdle"
+          "Mpd"
+          "NetworkBandwidth"
+          "Ping"
+          "Processes"
+          "Smb"
+          "Users"
+          "XIdleTime"
+          "XPath"
+        ]);
+        description = mdDoc ''
+          Name of the class implementing the check.  If this option is not specified, the check's name must represent a valid internal check class.
+        '';
+      };
+    };
+
+  wakeupType = with types;
+    submodule {
+      freeformType = with types; attrsOf anything;
+
+      options.enabled = mkOption {
+        default = false;
+        type = bool;
+        description = mdDoc ''
+          Whether to execute this wake-up check.
+        '';
+      };
```

Comment:
Yeah I didn't use `mkEnableOption` because this isn't really a "nixos-style" `enable` option, but if you think it's better to use it here that's fine for me